src/Lib/BSP/Lumps/GameLumps/StaticPropLump.cs

```diff
@@ -0,0 +1,241 @@
+                    break;
+            }
+            // v7* only
+            if (ActualVersion == StaticPropVersion.V7s)
+            {
+                prop.FlagsV7s = reader.ReadUInt32();
+                prop.LightmapResX = reader.ReadUInt16();
+                prop.LightmapResY = reader.ReadUInt16();
+            }
+            // since v8
+            if (ActualVersion >= StaticPropVersion.V8)
+            {
+                prop.MinCPULevel = reader.ReadByte();
+                prop.MaxCPULevel = reader.ReadByte();
+                prop.MinGPULevel = reader.ReadByte();
+                prop.MaxGPULevel = reader.ReadByte();
+            }
+            // since v7
+            if (ActualVersion >= StaticPropVersion.V7)
+            {
+                byte r = reader.ReadByte();
+                byte g = reader.ReadByte();
+                byte b = reader.ReadByte();
+                byte a = reader.ReadByte();
+                prop.DiffuseModulation = Color.FromArgb(a, r, g, b);
+            }
+            // v9 and v10 only
+            // and v11?
+            if (ActualVersion >= StaticPropVersion.V9)
+                prop.DisableX360 = reader.ReadInt32() > 0;
+            // since v10
+            if (ActualVersion >= StaticPropVersion.V10)
+                prop.FlagsEx = reader.ReadUInt32();
+            // since v11
+            if (ActualVersion >= StaticPropVersion.V11)
+                prop.UniformScale = System.BitConverter.ToSingle(reader.ReadBytes(4));
+            Data.Add(prop);
+            if (reader.BaseStream.Position - startpos != StructureSize)
+                throw new InvalidDataException($"bad({reader.BaseStream.Position - startpos} != {StructureSize})");
+        }
+
+        protected override void WriteItem(BinaryWriter writer, int index)
+        {
+            long startpos = writer.BaseStream.Position;
+            var prop = Data[index];
+            writer.Write(prop.Origin.X);
+            writer.Write(prop.Origin.Y);
+            writer.Write(prop.Origin.Z);
+            writer.Write(prop.Angle.Pitch);
+            writer.Write(prop.Angle.Yaw);
+            writer.Write(prop.Angle.Roll);
+
+            // v4
+            writer.Write(prop.PropType);
+            writer.Write(prop.FirstLeaf);
+            writer.Write(prop.LeafCount);
+            writer.Write(prop.Solid);
+            // every version except v7*
+            //if (Version != StaticPropVersion.V7s)
+            writer.Write(prop.Flags);
+            // v4 still
+            writer.Write(prop.Skin);
+            writer.Write(prop.FadeMinDist);
+            writer.Write(prop.FadeMaxDist);
+            writer.Write(prop.LightingOrigin.X);
+            writer.Write(prop.LightingOrigin.Y);
+            writer.Write(prop.LightingOrigin.Z);
+            // since v5
+            if (ActualVersion >= StaticPropVersion.V5)
+                writer.Write(prop.ForcedFadeScale);
+            // v6, v7, and v7* only
+            switch (ActualVersion)
```

Comment:
No default case?

Reply:
yes it's just like all the other ifs there .. it's a switch so have to check 3 things

Reply:
Then it can be condensed to one if statement if it's just these three coalescing into one function call.